temporian/core/event_set_ops.py

```diff
@@ -2133,6 +2133,85 @@ def cast(
 
         return cast(self, target=target, check_overflow=check_overflow)
 
+    def cumprod(
+        self: EventSetOrNode,
+        sampling: Optional[EventSetOrNode] = None,
+    ) -> EventSetOrNode:
+        """Computes the cumulative product of values over each feature in an
+        [`EventSet`][temporian.EventSet].
+
+        This operation only supports floating-point features.
+
+        Foreach timestamp, calculate the product of the feature from the beginning.
+        Shorthand for `moving_product(event, window_length=np.inf)`.
+
+        Missing (NaN) values are not accounted for. The output will be NaN until
+        the input contains at least one numeric value.
+
+        If `sampling` is specified or `window_length` is an EventSet, the moving
+        window is sampled at each timestamp in them, else it is sampled on the
+        input's.
+
+        Warning: The `cumprod` function leverages an infinite window length for
```

Comment:
Again, no need to explain the underlying implementation but **do** keep the warning